frontend/src/bundles/auth/components/sign-up-form/sign-up-form.tsx

```diff
@@ -1,26 +1,30 @@
+import { Link } from 'react-router-dom';
+
 import {
     Button,
     ButtonSize,
     ButtonVariant,
     Input,
+    Loader,
 } from '~/bundles/common/components/components.js';
+import { AppRoute } from '~/bundles/common/enums/app-route.enum.js';
 import { useAppForm, useCallback } from '~/bundles/common/hooks/hooks.js';
-import {
-    type UserAuthRequestDto,
-    userAuthValidationSchema,
-} from '~/bundles/users/users.js';
+import { userSignUpValidationSchema } from '~/bundles/users/users.js';
 
 import { DEFAULT_SIGN_UP_PAYLOAD } from './constants/constants.js';
+import { type UserSignUpForm } from './type.js';
 
 type Properties = {
-    onSubmit: (payload: UserAuthRequestDto) => void;
+    onSubmit: (payload: UserSignUpForm) => void;
+    isLoading: boolean;
 };
 
-const SignUpForm: React.FC<Properties> = ({ onSubmit }) => {
-    const { control, errors, handleSubmit } = useAppForm<UserAuthRequestDto>({
-        defaultValues: DEFAULT_SIGN_UP_PAYLOAD,
-        validationSchema: userAuthValidationSchema,
-    });
+const SignUpForm: React.FC<Properties> = ({ onSubmit, isLoading }) => {
+    const { control, errors, isValid, handleSubmit } =
+        useAppForm<UserSignUpForm>({
+            defaultValues: DEFAULT_SIGN_UP_PAYLOAD,
+            validationSchema: userSignUpValidationSchema,
```

Comment:
```suggestion
            validationSchema: userSignUpValidationSchema,
            shouldUnregister: false,
```
shouldUnregister for prevent reset form after submit

Reply:
I don't undertand this, because signup form doesn't reset after submit

Reply:
I tested, really does not reset) Sorry, you can also remove it in sign-in form.